tests/nexus/json_generator_test.py

```diff
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+import scippnexus as snx
+from scipp.testing import assert_identical
+
+from ess.reduce import data
+from ess.reduce.nexus.json_generator import event_data_generator
+from ess.reduce.nexus.json_nexus import json_nexus_group
+
+
+def test_event_data_generator_monitor_events_round_trip() -> None:
+    filename = data.loki_tutorial_sample_run_60250()
+    monitor = snx.load(filename, root='entry/instrument/monitor_1/monitor_1_events')
+    generator = event_data_generator(monitor)
+    for i in range(len(monitor)):
+        group = json_nexus_group(next(generator))
+        assert_identical(group[()], monitor[i : i + 1])
```

Comment:
Can you add a test that the generator has finished after the loop? I.e., that `next(generator)` raises `StopIteration`. Asking because the loop uses `monitor` and not `generator`.